sdk/src/wallet/operations/transaction/high_level/delegation/create.rs

```diff
@@ -0,0 +1,194 @@
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let options = options.into();
+        let prepared_transaction = self.prepare_create_delegation_output(params, options.clone()).await?;
+
+        self.sign_and_submit_transaction(prepared_transaction, None, options)
+            .await
+    }
+
+    /// Prepares the transaction for [Wallet::create_delegation_output()].
+    pub async fn prepare_create_delegation_output(
+        &self,
+        params: CreateDelegationParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<PreparedTransactionData> {
+        log::debug!("[TRANSACTION] prepare_create_delegation_output");
+        let protocol_parameters = self.client().get_protocol_parameters().await?;
+
+        let address = match params.address.as_ref() {
+            Some(bech32_address) => {
+                self.client().bech32_hrp_matches(bech32_address.hrp()).await?;
+                bech32_address.inner().clone()
+            }
+            None => self.address().await.inner().clone(),
+        };
+
+        let mut options = options.into();
+        let slot_commitment_id = if let Some(commitment_context) = options
+            .iter()
+            .flat_map(|opts| opts.context_inputs.iter())
+            .flat_map(|c| c.iter())
+            .find_map(|c| c.as_commitment_opt())
+        {
+            commitment_context.slot_commitment_id()
+        } else {
+            // Add a commitment context input with the latest slot commitment
+            let latest_id = self.client().get_info().await?.node_info.status.latest_commitment_id;
+            let context_input = ContextInput::from(CommitmentContextInput::new(latest_id));
+            if let Some(options) = &mut options {
+                if let Some(context_inputs) = &mut options.context_inputs {
+                    context_inputs.push(context_input);
+                } else {
+                    options.context_inputs.replace(vec![context_input]);
+                }
+            } else {
+                options.replace(TransactionOptions {
+                    context_inputs: Some(vec![context_input]),
+                    ..Default::default()
+                });
+            }
+            latest_id
+        };
+
+        let past_bounded_slot_index = slot_commitment_id
+            .slot_index()
+            .past_bounded_slot(protocol_parameters.max_committable_age);
+        let past_bounded_epoch_index =
+            past_bounded_slot_index.to_epoch_index(protocol_parameters.slots_per_epoch_exponent);
+
+        let registration_slot = (past_bounded_epoch_index + 1).registration_slot(
+            protocol_parameters.slots_per_epoch_exponent,
+            protocol_parameters.epoch_nearing_threshold,
+        );
+
+        let delegation_output_builder = DelegationOutputBuilder::new_with_minimum_amount(
+            protocol_parameters.storage_score_parameters(),
```

Comment:
Use the delegated_amount for the output amount, they must be equal at creation, probably also answers the TODO at L26

Reply:
I made this happen in the `DelegationOutputBuilder::finish` method. I don't know if it's correct to do that, but I can't see a scenario where these values are not the same 🤷‍♀️

Reply:
Ah missed that